Reducers/Reducer.js

```diff
@@ -0,0 +1,78 @@
+//todo get inital state from storage
+//todo implement import/export
+//todo learn functional programming. //per Ben. 
+
+const initialState = {
+    characters: [
+      {
+        id: 1,
+        name: 'Cohen',
+        race: 'Human',
+        classes: [
+          {
+            name: 'Barbarian',
+            level: 2,
+          },
+          {
+            name: 'Cleric',
+            level: 3,
+          },
+        ],
+        hitPoints: {
+          current: 28,
+          max: 28,
+        },
+        armorClass: 16,
+        speed: 30,
+        abilities: {
+          dex: { score: 13 },
+          wis: { score: 8 },
+        },
+        proficiencyBonus: 3,
+        skills: {
+          per: { isProficient: true },
+        },
+        characterAbilities: {
+          str: { name: 'Strength', score: 8, isProficient: true },
+          dex: { name: 'Dexterity', score: 13, isProficient: false },
+          con: { name: 'Constitution', score: 8, isProficient: true },
+          int: { name: 'Intelligence', score: 16, isProficient: false },
+          wis: { name: 'Wisdom', score: 8, isProficient: false },
+          cha: { name: 'Charisma', score: 6, isProficient: false },
+        }
+      }
+    ]    
+  };
+
+const levelUp = (curState, action) => {
+  return {...curState,
+    characters: curState.characters.map(c => {
+      if(action.id === c.id){
+        return {...c, 
+          classes: c.classes.map(cl => ({
+            ...cl,
+            level: cl.level + 1
+          }))
+        }
+      }
+      return c;
+    })
+  }
+}
```

Comment:
This is a reducer. Reducers have to meet certain requirements, like not mutating state. Rule of thumb, if you have an = in your reducer then something is wrong. There are libraries that make this syntax easier to read.